src/go/k8s/controllers/redpanda/console_controller.go

```diff
@@ -71,9 +74,13 @@ func (r *ConsoleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 
 	var cluster redpandav1alpha1.Cluster
 	if err := r.Get(ctx, types.NamespacedName{
-		Namespace: console.Spec.ClusterKeyRef.Namespace,
-		Name:      console.Spec.ClusterKeyRef.Name,
+		Namespace: console.GetNamespace(),
+		Name:      console.GetName(),
 	}, &cluster); err != nil {
+		if apierrors.IsNotFound(err) {
+			// WANT: We should probably update status.condition here.
+			return ctrl.Result{}, nil
+		}
```

Comment:
I agree we should. You can add condition as Nicola did in cluster CR. 

REF https://github.com/redpanda-data/redpanda/pull/3978

Reply:
Add conditions in `ConsoleStatus`